src/App.tsx

```diff
@@ -1,35 +1,52 @@
-import React from 'react';
+import React, { ChangeEvent } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
+import { SearchBar } from './components/SearchBar';
 
-type State = {};
+type Movie = {
+  title: string;
+  description: string;
+  imgUrl: string;
+  imdbUrl: string;
+  imdbId: string;
+};
+
+type State = {
+  query:string;
+  moviesList:Movie[];
+};
 
 export class App extends React.Component<{}, State> {
-  state: State = {};
+  state: State = {
+    query: '',
+    moviesList: moviesFromServer,
+  };
+
+  handleChangeEvent = (event:ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+    const initialMoviesList = moviesFromServer;
+
+    this.setState({
+      query: value.toLowerCase(),
+    });
+
+    this.setState({
+      moviesList: initialMoviesList.filter(
+        movie => {
+          return (movie.title.toLowerCase().includes(value)
+            || movie.description.toLowerCase().includes(value));
```

Comment:
do not take away the user’s ability to type UpperCase letters. Just do query to lowerCase before filtering (not in a state)